### DISCUSSION TITLE
Refetch when new ttl would override cached value

### DISCUSSION THREAD

**Discussion by jodoox** [arthurfiorette/axios-cache-interceptor#625] [Q&A] [answered]
Hi, thanks for your great job with this library. 

I'm wondering how to handle a case where I fetch data from an expensive endpoint with a 12-hour ttl, but I sometimes need to ensure the data is no more than 30-min old. 

My understanding is that the `ttl` argument at the query level will be saved along the query response, but that subsequent queries will not have their ttl checked against this "original" value- e.g. in the example below, the second call still returns the initial value. 

```
[t0] axios.get('/expensive_endpoint', { cache: {ttl: 12_HOURS_IN_MIL_SEC}}) --> resp_t0
[t0+5h]  axios.get('/expensive_endpoint', { cache: {ttl: 30_MIN_IN_MIL_SEC}}) --> resp_t0
```

Is there any way to control this with the library (i.e. compute the query key, check its cached `ttl` if any, compare it against the new one, invalidate the cached value and proceed with the new query if needed)? Setting `override: true` seems underoptimal as I wouldn't be caching anymore.

Thanks

**Comment by arthurfiorette** [comment, marked as answer]
Hey @jodoox, I think I understood your requirements correct. `override` seems to be a pretty good case of this usage. Override does not ignores cache completely, `cache: false` does. You could do the following:

```ts
const id = 'some-id';
const currentStorage = await axios.storage.get(id)
const isCacheOlderThan30Min = myLogic(currentStorage);
axios.get('url', { id, cache: { override: isCacheOlderThan30Min } });
```

This should only define  `override: true` when the current cache is older than 30min, in which case, as the override spec says, would ignore current cache and proceed to request in the network, however it does not deletes or ignore cache/fallbacks mechanisms, which is something you should care about to have.

However, if you do not have direct access to the request property, you could just do an `axios.storage.remove` call when you get that your query should not be kept for more than 30 min.

Also, as far as I understood, you still wanna keep the cache for 12 hours for some requests. Because, if not, you could just override the `headerInterpreter` to resolve every time the same request to be 30 min.

**Reply by jodoox** [reply]
Thanks for this, I'll have a try
